rosbag2_py/src/rosbag2_py/format_bag_metadata.cpp

```diff
@@ -42,9 +42,10 @@ std::unordered_map<std::string, std::string> format_duration(
   std::chrono::high_resolution_clock::duration duration)
 {
   std::unordered_map<std::string, std::string> formatted_duration;
-  auto m_seconds = std::chrono::duration_cast<std::chrono::milliseconds>(duration);
-  auto seconds = std::chrono::duration_cast<std::chrono::seconds>(m_seconds);
-  std::string fractional_seconds = std::to_string(m_seconds.count() % 1000);
+  auto seconds = std::chrono::duration_cast<std::chrono::seconds>(duration);
+  auto nanoseconds = std::chrono::duration_cast<std::chrono::nanoseconds>(duration);
+  auto nanoseconds_from_seconds = std::chrono::duration_cast<std::chrono::nanoseconds>(seconds);
+  std::string fractional_seconds = std::to_string((nanoseconds - nanoseconds_from_seconds).count());
```

Comment:
I think this might still have the bug where leading zeros are lost.

If `duration` is 1 second and 1 nanosecond, `seconds` will equal `1`, `nanoseconds` will equal `1000000001`, and `nanoseconds_from_seconds` will equal `1000000000`.

`nanoseconds - nanoseconds_from_seconds` = `1000000001 - 1000000000` == `1`, and `fractional_seconds` = `std::to_string(1)` which is `"1"`, but the answer we really want is `"000000001"`.

Reply:
@sloretz Yeah, you are right. I've fixed it by using 
```
fractional_seconds_ss << std::setw(9) << std::setfill('0') << (nanoseconds - nanoseconds_from_seconds).count();
```

Reply:
Hmm, integration tests failed. Seems need to make adjustments.
I will take a look.

Reply:
Adjusted expectations in info end-to-end tests by adding leading zero

The real file duration of the recording with sqlite3 is: `70633730` nanoseconds. (from metadata.yaml file)
i.e., regex mask shall be `"0\\.0706.*s"` to match `070633730` nanoseconds. 9 digits for nano.
It was a wrong expectation in tests.